src/tactic/core.lean

```diff
@@ -1033,6 +1021,51 @@ meta def mk_meta_pis : expr → tactic (list expr × expr)
   return ((p :: ps), r)
 | e := return ([], e)
 
+end tactic
+
+namespace lean.parser
+open lean interaction_monad.result
+
+/-- `emit_command_here str` behaves as if the string `str` were placed as a user command at the
+current line. -/
+meta def emit_command_here (str : string) : lean.parser string :=
+do (_, left) ← with_input command_like str,
+   return left
+
+/-- `emit_code_here str` behaves as if the string `str` were placed at the current location in
+source code. -/
+meta def emit_code_here : string → lean.parser unit
+| str := do left ← emit_command_here str,
+            if left.length = 0 then return ()
+            else emit_code_here left
+
+/-- `get_current_namespace` returns the current namespace (it could be `name.anonymous`).
+
+This function deserves a C++ implementation in core lean, and will fail if it is not called from
+the body of a command (i.e. anywhere else that the `lean.parser` monad can be invoked). -/
+meta def get_current_namespace : lean.parser name :=
```

Comment:
Is there an issue open for this/are @gebner @cipher1024 aware?

I'm happy to merge this now and replace it once this lands in core. It's not that bad of a hack.

Reply:
Yes there is: https://github.com/leanprover-community/lean/issues/196